buildpacktoml_test.go

```diff
@@ -0,0 +1,773 @@
+package lifecycle_test
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/google/go-cmp/cmp"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+
+	"github.com/buildpacks/lifecycle"
+	"github.com/buildpacks/lifecycle/api"
+	"github.com/buildpacks/lifecycle/env"
+	"github.com/buildpacks/lifecycle/launch"
+	"github.com/buildpacks/lifecycle/layers"
+	h "github.com/buildpacks/lifecycle/testhelpers"
+	"github.com/buildpacks/lifecycle/testmock"
+)
+
+var latestBuildpackAPI = api.Buildpack.Latest()
+
+func TestBuildpackTOML(t *testing.T) {
+	spec.Run(t, "BuildpackTOML", testBuildpackTOML, spec.Report(report.Terminal{}))
+}
+
+//go:generate mockgen -package testmock -destination testmock/env.go github.com/buildpacks/lifecycle BuildEnv
+
+func testBuildpackTOML(t *testing.T, when spec.G, it spec.S) {
+	var (
+		bpTOML         lifecycle.BuildpackTOML
+		mockCtrl       *gomock.Controller
+		mockEnv        *testmock.MockBuildEnv
+		stdout, stderr *bytes.Buffer
+		tmpDir         string
+		platformDir    string
+		appDir         string
+		layersDir      string
+		buildpacksDir  string
+		config         lifecycle.BuildConfig
+	)
+
+	it.Before(func() {
+		mockCtrl = gomock.NewController(t)
+		mockEnv = testmock.NewMockBuildEnv(mockCtrl)
+
+		var err error
+		tmpDir, err = ioutil.TempDir("", "lifecycle")
+		if err != nil {
+			t.Fatalf("Error: %s\n", err)
+		}
+		stdout, stderr = &bytes.Buffer{}, &bytes.Buffer{}
+		platformDir = filepath.Join(tmpDir, "platform")
+		layersDir = filepath.Join(tmpDir, "launch")
+		appDir = filepath.Join(layersDir, "app")
+		mkdir(t, layersDir, appDir, filepath.Join(platformDir, "env"))
+
+		buildpacksDir, err = filepath.Abs(filepath.Join("testdata", "by-id"))
```

Comment:
I was going to suggest a cleanup of the testdata here given we don't need the full symlinked buildpacks dir for this test, but I am realizing it is probably needed for detect tests. This is something we could likely simplify in the future when we refactor detector to use the new patterns.